python/ray/serve/_private/replica.py

```diff
@@ -371,17 +372,36 @@ async def reconfigure(
         deployment_config: DeploymentConfig,
     ) -> Tuple[DeploymentConfig, DeploymentVersion]:
         try:
-            await self._user_callable_wrapper.reconfigure(deployment_config)
-            return await self._get_metadata()
+            user_config_changed = (
+                deployment_config.user_config != self._deployment_config.user_config
+            )
+            logging_config_changed = (
+                deployment_config.logging_config
+                != self._deployment_config.logging_config
+            )
+            self._deployment_config = deployment_config
+            self._version = DeploymentVersion.from_deployment_version(
+                self._version, deployment_config
+            )
+
+            if logging_config_changed:
+                self._configure_logger_and_profilers(deployment_config.logging_config)
```

Comment:
@sihanwang41 this is actually fixing a latent bug -- when I updated the replica log format, it was not updated in the `logging_config` update path because they were completely disjoint. This is an example of why on the original PR I asked if we could unify where we're setting logging config every where if possible.

Reply:
Yes let me see what I can do. Probably just need to assert something about the log format in the existing tests.

Reply:
Ok I've added assertions to `test_deploy_app`'s logging test that cover this.